test/integration/boundingbox_camera_plugin.cc

```diff
@@ -0,0 +1,474 @@
+  ignition::rendering::unloadEngine(engine->Name());
+}
+
+/////////////////////////////////////////////////
+void BoundingBoxCameraSensorTest::Boxes3DWithBuiltinSDF(
+  const std::string &_renderEngine)
+{
+  std::string path = ignition::common::joinPaths(PROJECT_SOURCE_PATH, "test",
+      "integration", "boundingbox_3d_camera_sensor_builtin.sdf");
+  sdf::SDFPtr doc(new sdf::SDF());
+  sdf::init(doc);
+  ASSERT_TRUE(sdf::readFile(path, doc));
+  ASSERT_NE(nullptr, doc->Root());
+  ASSERT_TRUE(doc->Root()->HasElement("model"));
+  auto modelPtr = doc->Root()->GetElement("model");
+  ASSERT_TRUE(modelPtr->HasElement("link"));
+  auto linkPtr = modelPtr->GetElement("link");
+  ASSERT_TRUE(linkPtr->HasElement("sensor"));
+  auto sensorPtr = linkPtr->GetElement("sensor");
+  ASSERT_TRUE(sensorPtr->HasElement("camera"));
+  auto cameraPtr = sensorPtr->GetElement("camera");
+  ASSERT_TRUE(cameraPtr->HasElement("image"));
+  auto imagePtr = cameraPtr->GetElement("image");
+
+  unsigned int width = imagePtr->Get<int>("width");
+  unsigned int height = imagePtr->Get<int>("height");
+
+  EXPECT_EQ(width, 320u);
+  EXPECT_EQ(height, 240u);
+
+  // Setup ign-rendering with an empty scene
+  auto *engine = ignition::rendering::engine(_renderEngine);
+  if (!engine)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' is not supported" << std::endl;
+    return;
+  }
+
+  ignition::rendering::ScenePtr scene = engine->CreateScene("scene");
+  BuildScene3D(scene);
+
+  ignition::sensors::Manager mgr;
+
+  sdf::Sensor sdfSensor;
+  sdfSensor.Load(sensorPtr);
+
+  std::string type = sdfSensor.TypeStr();
+  EXPECT_EQ(type, "boundingbox_camera");
+
+  ignition::sensors::BoundingBoxCameraSensor *sensor =
+    mgr.CreateSensor<ignition::sensors::BoundingBoxCameraSensor>(sdfSensor);
+
+  EXPECT_NE(sensor, nullptr);
+  sensor->SetScene(scene);
+
+  EXPECT_EQ(width, sensor->ImageWidth());
+  EXPECT_EQ(height, sensor->ImageHeight());
+
+  auto camera = sensor->BoundingBoxCamera();
+  EXPECT_NE(camera, nullptr);
+
+  camera->SetLocalPosition(0.0, 0.0, 0.0);
+  camera->SetLocalRotation(0.0, 0.0, 0.0);
+  camera->SetAspectRatio(1.333);
+  camera->SetHFOV(IGN_PI / 2);
+  camera->SetBoundingBoxType(BoundingBoxType::Box3D);
+
+  // Get the Msg
+  std::string topic =
+    "/test/integration/BoundingBox3DCameraPlugin_boxesWithBuiltinSDF";
+  WaitForMessageTestHelper<
+    ignition::msgs::AnnotatedOriented3DBox_V> helper(topic);
+
+  // Update once to create image
+  mgr.RunOnce(std::chrono::steady_clock::duration::zero());
+
+  EXPECT_TRUE(helper.WaitForMessage()) << helper;
+
+  // subscribe to the BoundingBox camera topic
+  ignition::transport::Node node;
+  node.Subscribe(topic, &OnNew3DBoundingBoxes);
+
+  // wait for a few BoundingBox camera boxes
+  mgr.RunOnce(std::chrono::steady_clock::duration::zero(), true);
+  // wait for a new boxes
+  WaitForNewFrame();
+
+  g_mutex.lock();
+  g_counter = 0;
+
+  // check the hidden box
```

Comment:
This comment seems out of place since there are no hidden boxes in the 3D boxes test. Can you remove it?